specs/monitoring_integration.adoc

```diff
@@ -0,0 +1,321 @@
+= Spec - Introducing a new component monitoring-integration
+
+Remove existing performance monitoring component and replace it with a new
+monitoring-integration component.
+
+
+== Problem description
+
+The new monitoring-integration component would primarily focus on dynamic
+configuration of grafana, based on the topology changes in etcd. Some of these
+responsibilities would be:
+
+* monitoring-integration should store predefined json’s for dashboards, panels and alerts.
+* Keeping track of the topology changes in etcd to configure the grafana dashboards
+  to ensure that the correct nodes are part of the respective cluster dashboards.
+* Configuring grafana authentication and authorization (https://github.com/Tendrl/specifications/pull/205),
+  based on changes in the primary tendrl AA scheme.
+
+
+== Use Cases
+
+* Monitoring-integration should be able to configure grafana dashboard and panels
+  and alerts.
+* User should be able access grafana dashboard via tendrl user interface using
+  tendrl authentication.
+* Grafana alerting events should be notified via tendrl.
+
+== Proposed change
+
+=== Why monitoring-integration?
+
+Grafana is used for visualizing time series data in graphs. Based on the metrics
+from datasource it will create graphs and display. Grafana supports different kinds
+of http api's to configure its environment.
+
+1. Admin
+2. Alerting
+3. Authentication
+4. Dashboard
+5. Dashboard version
+6. Data source
+7. Organization
+8. Users
+9. Preference
+10. Snapshot
+11. Others
+
+Monitoring integration is the component which will be used to configure the grafana
+environment when it is started. Monitoring integration is going to have all json
+for setting up the grafana environment for tendrl users. Monitoring integration should
+keep track of the topology changes in etcd to configure the grafana dashboards.
+
+Monitoring integration should have all grafana related packages as dependency.
+When cluster is created or imported, the flow will install the monitoring integration
```

Comment:
Monitoring integration is installed when the cluster imported/created? So you mean it is per cluster? I don't think this is correct. Per my understanding this service(grafana as well) is per tendrl instance(just like performance monitoring) and dash board poulation should be done as part of import
If that is the case, how monitoring integration is installed?(I assume via tendrl-ansible). Please provide the complete workflow here

@anivargi , @brainfunked , please comment